tools/dataset_converters/update_infos_to_v2.py

```diff
@@ -387,6 +388,142 @@ def update_nuscenes_infos(pkl_path, out_dir):
 
     mmengine.dump(converted_data_info, out_path, 'pkl')
 
+def update_tier4_dataset_infos(pkl_path, out_dir):
```

Comment:
@kaancolak Please make sure that the dataset name T4Dataset consistent **throughout the repository**, not only the variable names or function names but also the file names as well.